torch/distributed/tensor/_ops/_tensor_ops.py

```diff
@@ -653,12 +684,20 @@ def stack_strategy(op_schema: OpSchema) -> StrategyType:
 
     follow_placements = normalize_shard_for_stack(follow_placements, dim)
 
-    op_strategy.strategies.append(
-        OpSpec(
-            output_specs=DTensorSpec(mesh, tuple(follow_placements)),
-            input_specs=input_specs,
+    for strategy in input_tuple_strategy.childs:
+        assert isinstance(strategy, OpStrategy)
+        output_spec = DTensorSpec(mesh, tuple(follow_placements))
+        redistribute_cost = []
+        for input_spec in input_specs:
```

Comment:
i'm having trouble following the change here.

1- it looks like we decide one common strategy to force on all our inputs and thus also our output.  This is the strategy based on the follow_placements.
2- once we decide this, i expect that we will have one call to `generate_redistribute_costs` per input tensor.  This is because each input tensor has a single destination spec: the one based on follow_placements.

instead,  there is a  for loop over input_specs.  note: each input in input_specs is identical, so each cost generated here is also identical.  Is the duplication helpful/necessary?